File: Problems/No.1480_Running_Sum_of_1d_Array/running_sum_of_1d_array.py
class Solution(object):
    def runningSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        # parameter
        Rsum = 0
        Running_Sum = []

        # start computing
        for num in nums:
            Rsum += num
            Running_Sum.append(Rsum)
        
        return Running_Sum

File: Problems/No.1480_Running_Sum_of_1d_Array/running_sum_of_1d_array.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        
        // parameter
        std::vector<int> Running_Sum(nums.size(), 0);

        // start computing
        std::partial_sum(nums.begin(), nums.end(), Running_Sum.begin());

        return Running_Sum;

    }
};
